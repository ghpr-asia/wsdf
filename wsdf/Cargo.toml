[package]
name = "wsdf"
version = "0.1.0"
edition = "2021"
autotests = false
license = "MIT OR Apache-2.0"
description = "Wireshark Dissector Framework"
homepage = "https://github.com/ghpr-asia/wsdf"
documentation = "https://docs.rs/wsdf"
repository = "https://github.com/ghpr-asia/wsdf"
readme = "README.md"
keywords = ["wireshark", "dissectors"]
categories = ["network-programming"]

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dev-dependencies]
trybuild = { version = "1.0", features = ["diff"] }

[dependencies]
wsdf-derive = { path = "../wsdf-derive", version = "0.1.0" }
epan-sys = { path = "../epan-sys", version = "0.1.0" }
thiserror = "1.0"

[[example]]
name = "udp"
crate-type = ["cdylib"]

[[example]]
name = "moldudp64"
crate-type = ["cdylib"]

[[example]]
name = "dns"
crate-type = ["cdylib"]
