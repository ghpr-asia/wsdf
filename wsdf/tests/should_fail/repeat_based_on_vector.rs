// Tests what happens if we try to have a vector repeat on a vector?

use wsdf::*;

#[derive(Protocol)]
#[wsdf(decode_from = "moldudp.payload")]
struct ProtoFoo {
    n: u32,
    #[wsdf(len_field = "n")]
    xs: Vec<u32>,
    #[wsdf(len_field = "xs")]
    ys: Vec<u32>,
}

fn main() {}
