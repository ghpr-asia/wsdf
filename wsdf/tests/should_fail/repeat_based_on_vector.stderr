error: this field cannot be used to indicate length
  --> tests/should_fail/repeat_based_on_vector.rs:10:5
   |
10 |     xs: Vec<u32>,
   |     ^^
